trp_project_scrum/models/project_task_work.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api, _
+
+class ProjectTaskWork(models.Model):
+    _inherit = "project.task.work"
+
+
+    @api.model
+    def get_user_related_details(self, user_id):
+        emp_obj = self.env['hr.employee']
+        user =  self.env['res.users'].browse(
+            self.env.context.get('user_id') or self._uid
+        )
+        emp_id = emp_obj.search(
+            [('user_id','=', user.id)],
+            limit=1
+        )
+        if not emp_id and user.is_collaborator :
+            # defaults for therp
+            product_id = self.sudo().env['product.template'].search(
+                [
+                    ('type', '=', 'service'),
+                    ('uos_id.category_id.name' , '=' , 'Working Time')
+                ],
+                limit=1
+            )
+            res= {
+                'product_id' : product_id.id,
+                'journal_id' : self.sudo().env['account.analytic.journal'].search(
+                    [
+                        ('name', '=', 'Timesheet Journal'),
+                        ('code', '=', 'TS')
+                    ],
+                    limit = 1
+                ).id,
+                'general_account_id' :
+                    product_id.categ_id.property_account_expense_categ.id,
+                'product_uom_id' : product_id.uom_id.id
+            }
+            # TODO create default data for collaborator
+            """
+            res= {
+                'product_id' : self.env.ref('collaborator_default_product').id,
+                'journal_id' : self.env.ref('collaborator_default_journal').id,
+                'general_account_id' : self.env.ref(
+                    'collaborator_default_account'
+                ),
+                'product_uom_id' : self.env.ref(
+                    'collaborator_default_product'
+                ).uom_id.id,
+            }
+            """
+
+            return res
+
+        res = super(ProjectTaskWork, self).get_user_related_details()
+        return res
+
+
+    """
+    TODO
+    No timesheets for collaborators
+    """
+    """
+    @api.model
+    def _create_analytic_entries(self, vals):
+        user_model = self.env['res.users']
+        user_or_sudo = 1 if user_model.browse(
+            vals.get('user_id')
+        ).is_collaborator else self._uid
+        res=super(
+            ProjectTaskWork, self
+        )._create_analytic_entries(vals=vals)
+        return res
+    """
+
+    @api.model
+    def create(self, vals):
+        res = super(
+            ProjectTaskWork, self.with_context(no_analytic_entry=True)
```

Comment:
because the creation of the analytic account is triggered on creation when project_timesheet is installed. (it overwrites create too) with this key we tell him we do not want that. oh. but it should be conitional to is_sollaborator. fixing.